### PR TITLE
feat(calendar): enhanced calendar with month granularity support

### PR DESCRIPTION
**Pull request by diegosps**
This PR implements the #212 

To avoid any breaking change and weird names, I created `EnhancedCalendar`,`EnhancedCalendarCanvas`, `ResponsiveEnhancedCalendar` and `ResponsiveEnhancedCalendarCanvas` which supports `granularity`, `breakpoint` and `weekDirection`. The previous set of Calendar components stayed almost the same.

In this set of components, `yearSpacing`, `yearLegend`, `yearLegendPosition` and `yearLegendOffset` have been renamed to `blockSpacing`, `blockLegend`, `blockLegendPosition` and `blockLegendOffset`.

While `yearLegend` receives `year: number`, `blockLegend` receives:
```typescript
block: {
firstMonth: number
firstYear: number
lastMonth: number
lastYear: number
}

```

There are some examples of the new functionalities as new stories in the storybook.

I tried my best to put shared code in a single file and specific code in different trees. This way, tree shaking should work and users of the normal `Calendar` should not see any increase in bundle size.

Please, review thoroughly and check `index.d.ts`

Cya!!

### CODE REVIEW

**Comment by igorbalteiro** [comment]
Hi @plouc , how are you?

Is there any expectation of when this PR will be merged?

File: packages/calendar/index.d.ts [hunk 1]
```diff
@@ -95,15 +111,37 @@ declare module '@nivo/calendar' {
...
         CalendarCommonProps &
+        YearLegend &
+        Partial<{
+            onClick: (datum: CalendarDayData, event: React.MouseEvent<SVGRectElement>) => void
+        }>
+
+    export type EnhancedCalendarSvgProps = CalendarData &
+        CalendarCommonProps &
+        EnhancedCalendarCommonProps &
+        YearLegend &
```
Comment (JaredPh) [inline packages/calendar/index.d.ts:130]:
```suggestion
        BlockLegend &
```

**Comment by scherler** [comment]
image
It does not address the underlying issue to limit the month to display.

**Comment by wyze** [comment]
Thanks for this @diegosps but I decided to go with the implementation in #1503.

**Pull request closed without merging**